cmd/create/execution.go

```diff
@@ -195,16 +200,22 @@ func createExecutionCommand(ctx context.Context, args []string, cmdCtx cmdCore.C
 	case Recover:
 		return recoverExecution(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName)
 	case Task:
-		if executionRequest, err = createExecutionRequestForTask(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName); err != nil {
+		executionRequest, err = createExecutionRequestForTask(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName)
+		if err != nil {
 			return err
 		}
 	case Workflow:
-		if executionRequest, err = createExecutionRequestForWorkflow(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName); err != nil {
+		executionRequest, err = createExecutionRequestForWorkflow(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName)
+		if err != nil {
 			return err
 		}
 	default:
 		return fmt.Errorf("invalid execution type %v", execParams.execType)
 	}
+
+	if executionConfig.ClusterPool != "" {
+		executionRequest.Spec.ClusterAssignment = &admin.ClusterAssignment{ClusterPoolName: executionConfig.ClusterPool}
+	}
```

Comment:
Can we contain this change to createExecutionRequest. Modifying the create objected in places outside it initial creation becomes confusing